eradiate/scenes/biosphere.py

```diff
@@ -436,308 +794,112 @@ def __str__(self):
-            Reflectance spectrum of the leaves in the cloud. Must be a reflectance
-            spectrum (dimensionless). Default: 0.5.
-
-        Parameter ``leaf_transmittance`` (float or :class:`~eradiate.scenes.spectra.Spectrum`):
-            Transmittance spectrum of the leaves in the cloud. Must be a
-            transmittance spectrum (dimensionless). Default: 0.5.
-
-            Unit-enabled field (default units: cdu[length])
-        """
+        # Pre-process dict: apply units to unit-enabled fields
+        d_copy = copy(d)
 
-        if not os.path.exists(file_path):
-            raise FileNotFoundError(f"No file at {file_path} found.")
+        for field in cls._fields_supporting_units():
+            # Fetch user-specified unit if any
+            try:
+                field_units = d_copy.pop(f"{field}_units")
+            except KeyError:
+                # If no unit is specified, don't attempt conversion and let the
+                # constructor take care of it
+                continue
 
-        positions_ = []
-        orientations_ = []
-        with open(os.path.abspath(file_path), "r") as definition_file:
-            for i, line in enumerate(definition_file):
-                values = line.split()
+            # If a unit is found, try to apply it
+            # Bonus: if a unit field *and* a quantity were found, we convert the
+            # quantity to the unit
+            field_value = d_copy[field]
+            d_copy[field] = ensure_units(field_value, field_units, convert=True)
 
-                if i == 0:
-                    radius = ensure_units(float(values[0].strip()), "meter")
+        size = d_copy.pop("size")
+        if "leaf_definition" in d_copy:
+            leaf_cloud =  LeafCloud.from_rami(**d_copy)
+        else:
+            if d_copy["shape_type"] != "cube":
+                raise ValueError("The HomogeneousDiscreteCanopy must be used"
+                                 "with a cuboid leaf cloud.")
+            leaf_cloud = LeafCloud.from_parameters(**d_copy)
 
-                position = [
-                    float(values[1].strip()),
-                    float(values[2].strip()),
-                    float(values[3].strip()),
-                ]
+        return cls(leaf_cloud_specs=[(leaf_cloud, [[0, 0, 0]])], size=size)
 
-                positions_.append(position)
-                normal = [
-                    float(values[4].strip()),
-                    float(values[5].strip()),
-                    float(values[6].strip()),
-                ]
 
-                orientations_.append(normal)
+@BiosphereFactory.register("floating_spheres_canopy")
+@parse_docs
+@attr.s
+class FloatingSpheresCanopy(DiscreteCanopy):
+    """Generates the canopy for the `floating sphere scenarios<https://rami-benchmark.jrc.ec.europa.eu/HTML/RAMI3/EXPERIMENTS3/HETEROGENEOUS/FLOATING_SPHERES/SOLAR_DOMAIN/DISCRETE/DISCRETE.php>`_
+    in the RAMI benchmark"""
```

Comment:
Please add a one-sentence description of the canopy (see what I suggested for the other one)